app/experimenter/kinto/tasks.py

```diff
@@ -112,22 +86,60 @@ def nimbus_check_kinto_push_queue():
             application=application,
             is_end_requested=True,
         )
-        if queued_experiments.exists():
-            nimbus_push_experiment_to_kinto.delay(queued_experiments.first().id)
-            metrics.incr(
-                f"check_kinto_push_queue.{collection}_queued_experiment_selected"
+
+        if kinto_client.has_pending_review():
+            metrics.incr(f"check_kinto_push_queue.{collection}_pending_review")
+            continue
+        elif rejected_collection_data:
+            nimbus_handle_rejected_review.delay(collection, rejected_collection_data)
+        elif queued_experiments.exists():
+            nimbus_handle_launch_experiment.delay(
+                collection, queued_experiments.first().id
             )
         elif end_requested_experiments.exists():
-            nimbus_end_experiment_in_kinto.delay(end_requested_experiments.first().id)
-            metrics.incr(
-                f"check_kinto_push_queue.{collection}_end_requested_experiment_deleted"
+            nimbus_handle_end_experiment.delay(
+                collection, end_requested_experiments.first().id
             )
         else:
             metrics.incr(f"check_kinto_push_queue.{collection}_no_experiments_queued")
 
     metrics.incr("check_kinto_push_queue.completed")
 
 
+@app.task
+def nimbus_handle_rejected_review(collection, collection_data):
+    kinto_client = KintoClient(collection)
+    rejected_slug = kinto_client.get_rejected_record()
+    experiment = NimbusExperiment.objects.get(slug=rejected_slug)
+
+    if experiment.status == NimbusExperiment.Status.LIVE and experiment.is_end_requested:
+        experiment.is_end_requested = False
+    else:
+        experiment.status = NimbusExperiment.Status.DRAFT
+
+    experiment.save()
+
+    generate_nimbus_changelog(
+        experiment,
+        get_kinto_user(),
+        message=f'Rejected: {collection_data["last_reviewer_comment"]}',
+    )
+
+    kinto_client.rollback_changes()
+
+
+@app.task
+def nimbus_handle_launch_experiment(collection, experiment_id):
+    nimbus_push_experiment_to_kinto.delay(experiment_id)
+    metrics.incr(f"check_kinto_push_queue.{collection}_queued_experiment_selected")
+
+
+@app.task
+def nimbus_handle_end_experiment(collection, experiment_id):
+    nimbus_end_experiment_in_kinto.delay(experiment_id)
+    metrics.incr(f"check_kinto_push_queue.{collection}_end_requested_experiment_deleted")
```

Comment:
Don't need these, just rename the other tasks

Reply:
Can you elaborate?